query/src/main/scala/filodb/query/LogicalPlan.scala

```diff
@@ -10,6 +10,13 @@ sealed trait LogicalPlan {
     * It is false for RawSeriesLikePlan, MetadataQueryPlan, RawChunkMeta, ScalarTimeBasedPlan and ScalarFixedDoublePlan
     */
   def isRoutable: Boolean = true
+  def replaceFilters(filters: Seq[ColumnFilter]): LogicalPlan = {
+    this match {
+      case p: PeriodicSeriesPlan => p.replacePeriodicSeriesFilters(filters)
+      case r: RawSeriesLikePlan  => r.replaceRawSeriesFilters(filters)
+      case _                     => throw new UnsupportedOperationException("Logical plan not supported for update")
```

Comment:
Remove the default case. Since this is a sealed class, it should catch at compile time. Otherwise it defeats the purpose of adding methods this way - we will only catch missing implementations at runtime.